frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -349,34 +341,38 @@ func toCommand(ic instructions.Command, dispatchStatesByName map[string]*dispatc
 					}
 				}
 			} else {
-				if index < 0 || index >= len(allDispatchStates) {
-					return command{}, errors.Errorf("invalid stage index %d", index)
+				stn, err = allDispatchStates.findStateByIndex(index)
+				if err != nil {
+					return command{}, err
 				}
-				stn = allDispatchStates[index]
+				ds := *newDispatchStates()
+				ds.addState(stn)
+				cmd.sources = ds
```

Comment:
These lines seem to be no-op and always get overwritten

Reply:
Thanks, it's my mistake.